src/log.ts

```diff
@@ -959,13 +962,23 @@ class Log implements LogSeverityFunctions {
     entry: Entry | Entry[],
     opts?: WriteOptions | ApiResponseCallback
   ): Promise<ApiResponse> {
+    const isInfoAdded = getInstrumentationInfoStatus();
     const options = opts ? (opts as WriteOptions) : {};
+    // If instrumentation info was not added, means that this is first time
+    // log entry is written and that the instrumentation log entry could be
+    // generated for this request. If yes, then make sure we set partialSuccess, so entire
+    // request will make it through and only oversized entries will be dropped
+    if (!isInfoAdded) {
+      options.partialSuccess = true;
+    }
```

Comment:
I understand the reasoning but a question remains how does the library know that the instrumentation log entry was successfully ingested?

Reply:
Library doesn't know - we do send the log entry and rely on retries to deliver an entry. The entry containing instrumentation info supposed to be not oversized (e.g. library creates this entry and keep track on data added to the entry). partialSuccess is added to ensure that entry is not dropped if request contains oversized entries. I will take a look to see if we can also track the request success/failure and perhaps will add a logic to reset global flag to resend instrumentation data.